### PR TITLE
Avoiding authentication on every request with Guard

### PR DESCRIPTION
**Pull request by weaverryan**
Hi guys!

See symfony/symfony#27395 and symfony/symfony#27427

When we're happy with this, I can merge and handle the changes on 3.4, because it will be almost entirely different (with `supports()`, autowiring & the `Security` class - booya for progress!)

### CODE REVIEW

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by javiereguiluz** [comment]
@weaverryan just asking: is this also related to this old pending issue? https://github.com/symfony/symfony-docs/issues/5534

**Comment by weaverryan** [comment]
Hmm, it's not unfortunately :/ 

Let's merge this PR once the code PR's are merged, as this PR mentions changing your firewall to "stateless" to avoid this issue, which isn't actually true yet in the code.

**Pull request merged**
